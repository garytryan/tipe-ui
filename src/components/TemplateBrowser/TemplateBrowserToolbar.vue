<template>
  <div :data-tipe-ui="$options.name">
    <div class="filter-control-container">
      <tipe-template-browser-filter-control
        @change="$emit('filter-change', $event.value)"
      />
    </div>
    <div class="new-template-button-container">
      <tipe-button
        icon-before="plus"
        @click="$emit('new-template')"
      >
        New Template
      </tipe-button>
    </div>
  </div>
</template>

<script>
import TipeButton from '@/components/Button'
import TipeTemplateBrowserFilterControl from './TemplateBrowserFilterControl'

export default {
  name: 'TipeTemplateBrowserToolbar',
  components: {
    TipeButton,
    TipeTemplateBrowserFilterControl
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTemplateBrowserToolbar'] {
  height: 100%;
  width: 100%;
  display: grid;
  grid-template-columns: 1fr auto;
  grid-auto-flow: 100%;
}

.filter-control-container {
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
}

.new-template-button-container {
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
