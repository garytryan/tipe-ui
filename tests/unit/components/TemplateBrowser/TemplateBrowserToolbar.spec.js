import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import TemplateBrowserToolbar from '@/components/TemplateBrowser/TemplateBrowserToolbar'

describe('TemplateBrowserToolbar', () => {
  describe('<TemplateBrowserToolbar>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(TemplateBrowserToolbar)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TemplateBrowserToolbar)

      expect(wrapper.attributes()['data-tipe-ui']).toBe(
        'TipeTemplateBrowserToolbar'
      )
    })
  })
})
