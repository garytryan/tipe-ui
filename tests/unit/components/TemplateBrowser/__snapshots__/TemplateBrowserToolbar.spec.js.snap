// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateBrowserToolbar <TemplateBrowserToolbar> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeTemplateBrowserToolbar">
  <!---->
  <!---->
</div>
`;
